### ISSUE TITLE
What should we put in our invitations to new maintainers?

### ISSUE THREAD

**Issue by petertseng** [exercism/discussions#105]
As #97 has indicated to us, the power and duty to invite new maintainers rests in our (existing maintainers') hands.

But one big reason I had always been hesitant to do so in the past (instead preferring to let @kytrinyx do it) was because I didn't know what to say in my invitation to a new maintainer!

So @kytrinyx suggested having a template email for such an invitation, so that we don't have to keep making up emails all the time.

Before we get there, we might at least think of a list of topics that would go in such an email.
We may end up deciding that the list of topics is sufficient and leave it up to each person's own voice to craft an email from the list, but I believe that at least the list is useful.

Here's what I started with (based off of the initial invitation @kytrinyx sent me, as well)

* Thanks for your past contributions
* Would you like to be a maintainer? Docs at https://github.com/exercism/exercism.io/blob/master/docs/maintaining-a-track.md
* In short, helping others contribute like you have.
* Whatever you decide, you can keep on contributing as you have been of course!
* Thanks again

I would like to hear if someone has other items that should be added to the email.

If this line of work provides a useful result, we may consider doing such a thing for other related tasks. I don't have any in mind at the moment, but something to keep in mind.

When we have decided on either the list, or a fully-formed email, what should we do with it? Should we put it somewhere (perhaps https://github.com/exercism/exercism.io/blob/master/docs/maintaining-a-track.md is a good place for it!) or leave it in this issue?

**Comment by kytrinyx** [comment]
That list is exactly what I would want, which makes sense seeing as it's basically what I wrote to @petertseng when I invited him :)

> Thanks for your past contributions

For this if possible I'd like to be specific about what we like (being helpful in code reviews, being kind, meticulous code reviews—anything to positively reinforce behavior that we want to encourage more of, and would be delighted to see in other users.

@jtigger @parkerl you both have much better human-skills than I do, so if you have suggestions here, I'd love to hear them.

> When we have decided on either the list, or a fully-formed email, what should we do with it? 

Let's create a separate document for it in the `docs/` directory (next to the "maintaining a track" document). It's not something that a new maintainer would need to see immediately, so it doesn't necessarily make sense to have it in the same document, but it would be nice to know exactly where to look for it, so keeping it with the other documentation would be good.

**Comment by Insti** [comment]
Anecdata: When I was invited to be a maintainer on the xruby track, I'd already been active with PRs and reviews for a few weeks and @kotp asked me casually in the Gitter chat and I said yes. I was honoured to be asked.  If I'd received a big email with a whole lot of things I'm committing to for that would have been more scary. 

So I think I agree with @petertseng,  A shorter email that says 'we've noticed you and like what you're doing, will you join us?' and a resource in the repository explaining what, why and how.

**Comment by kytrinyx** [comment]
> So I think I agree with @petertseng, A shorter email that says 'we've noticed you and like what you're doing, will you join us?' and a resource in the repository explaining what, why and how.

Yeah, me too. Nothing too long, and nothing that indicates that you're committing to a whole lot more than you're already doing. We're asking you because you've already contributed in ways that we appreciated.

**Comment by petertseng** [comment]
Can I ask for clarification on:

> nothing that indicates that you're committing to a whole lot more than you're already doing

One could interpret this to mean one of two extremes:

* we believe the commitment is actually large, but we want to hide it for some reason
* we believe the commitment is actually small, and we want to make that clear

Though I think the answer is between the extremes: It seems that you (general "you", not any particular person I'm speaking to) have pretty big leeway in how much you choose to commit to in the new role. Your (@kytrinyx) invitation to me implied that, since you said **there's no pressure in how much or little to contribute**. So I think that is a good point to bring up.

> Let's create a separate document for it in the `docs/` directory (next to the "maintaining a track" document)

Okay. When I think I have learned what I can from this discussion (might be soon), I can start us off with a PR there.

**Comment by kytrinyx** [comment]
> pretty big leeway in how much you choose to commit to in the new role.

Yeah, absolutely. We can tap or nominate as many maintainers as necessary to keep a track active and humming along without overburdening any one person.

It should be up to each person how much they're willing to contribute. If a language track is getting less attention than it needs, that's not the fault of any individual, it's a systemic fault: we've not done enough (as a project) to attract contributors, mentor them, and nominate them to be maintainers.

> you said **there's no pressure in how much or little to contribute**. So I think that is a good point to bring up.

@petertseng Yeah, I think that's important.

**Comment by jtigger** [comment]
This discussion is landing in a spot that real resonates with how I've thought about it.

- short and sweet
- specific praise
- handing them the guilt-free quit card *immediately* (thinking: "quitting" is what's appropriate when you're done)
- making the scope of the commitment clear (perfectly done in [maintaining-a-track.md](https://github.com/exercism/exercism.io/blob/master/docs/maintaining-a-track.md)).

Here's a real-life example (this is the entire email):

> I'm a fan.  Katrina and Matt Morgan have also voiced their appreciation for your efforts and  attention to detail.
>
> Would you be interested in being an official maintainer of the Java track?  This is about enabling people who clearly show self-initiative.  The only commitment is the one you make with yourself. :)
>
> Here's our guide: https://github.com/exercism/exercism.io/blob/master/docs/maintaining-a-track.md
>
> Thoughts?

-

p.s. I think you've underrated your people skills, @kytrinyx.  When you're ready, feel free to start owning your competency in that subject area. ❤️

**Comment by petertseng** [comment]
Thank you for your input, everyone. Closing this issue doesn't mean that we are closed to new ideas, but simply that new ideas can now be made at https://github.com/exercism/exercism.io/pull/3329 and (after that PR is merged) with new PRs to that file.

**Comment by jtigger** [comment]
It occurs to me, re-reading the discussion and Peter's PR (well done, btw), that something is implied.  I'm not sure if it's worth calling out explicitly — this may all just go without saying.  It's this...

... that we look to attract maintainers that are self-starters and want to collaborate (i.e. specifically want to work with a group).  Your participation in Exercism is most satisfying if the participation itself is the reward.

We believe this is how we garner the best quality contributions for Exercism: to tap into people's internal motivations and not a promise for an external reward or some sense of obligation.

**Comment by kytrinyx** [comment]
That sounds about right, @jtigger. We really do want people to be doing this because they get something out of the process.
